File: dlna-dmr/src/main/java/com/android/cast/dlna/dmr/service/IRendererInterface.java
package com.android.cast.dlna.dmr.service;

import org.fourthline.cling.model.types.UnsignedIntegerTwoBytes;
import org.fourthline.cling.support.avtransport.AVTransportException;
import org.fourthline.cling.support.model.DeviceCapabilities;
import org.fourthline.cling.support.model.MediaInfo;
import org.fourthline.cling.support.model.PositionInfo;
import org.fourthline.cling.support.model.TransportAction;
import org.fourthline.cling.support.model.TransportInfo;
import org.fourthline.cling.support.model.TransportSettings;

/**
 *
 */
public interface IRendererInterface {

    interface IAVTransportControl {
        void setAVTransportURI(String currentURI, String currentURIMetaData) throws AVTransportException;

        void setNextAVTransportURI(String nextURI, String nextURIMetaData);

        void setPlayMode(String newPlayMode);

        void setRecordQualityMode(String newRecordQualityMode);

        void play(String speed) throws AVTransportException;

        void pause() throws AVTransportException;

        void seek(String unit, String target) throws AVTransportException;

        void previous();

        void next();

        void stop() throws AVTransportException;

        void record();

        TransportAction[] getCurrentTransportActions() throws Exception;

        DeviceCapabilities getDeviceCapabilities();

        MediaInfo getMediaInfo();

        PositionInfo getPositionInfo();

        TransportInfo getTransportInfo();

        TransportSettings getTransportSettings();
    }

    interface IAVTransportUpdate {
        void updateMediaCurrentPosition(long position);

        void updateMediaDuration(long duration);

        void updateMediaState(int state);
    }

    interface IAVTransport extends IAVTransportControl, IAVTransportUpdate {
    }

    // -------------------------------------------------------------------------------------------
    // - Audio
    // -------------------------------------------------------------------------------------------
    interface IAudioControl {
        void setMute(String channelName, boolean desiredMute);

        boolean getMute(String channelName);

        void setVolume(String channelName, UnsignedIntegerTwoBytes desiredVolume);

        UnsignedIntegerTwoBytes getVolume(String channelName);
    }
}

File: dlna-dmr/src/main/java/com/android/cast/dlna/dmr/service/IRendererInterface.kt
package com.android.cast.dlna.dmr.service

import org.fourthline.cling.model.types.UnsignedIntegerFourBytes
import org.fourthline.cling.model.types.UnsignedIntegerTwoBytes
import org.fourthline.cling.support.avtransport.AVTransportException
import org.fourthline.cling.support.model.*

/**
 *
 */
interface IRendererInterface {
    interface IControl {
        val instanceId: UnsignedIntegerFourBytes
    }

    // -------------------------------------------------------------------------------------------
    // - AvTransport
    // -------------------------------------------------------------------------------------------
    interface IAVTransportControl : IControl {
        @Throws(AVTransportException::class)
        fun setAVTransportURI(currentURI: String?, currentURIMetaData: String?)
        fun setNextAVTransportURI(nextURI: String?, nextURIMetaData: String?)
        fun setPlayMode(newPlayMode: String?)
        fun setRecordQualityMode(newRecordQualityMode: String?)

        @Throws(AVTransportException::class)
        fun play(speed: String?)

        @Throws(AVTransportException::class)
        fun pause()

        @Throws(AVTransportException::class)
        fun seek(unit: String?, target: String?)
        fun previous()
        operator fun next()

        @Throws(AVTransportException::class)
        fun stop()
        fun record()

        @get:Throws(Exception::class)
        val currentTransportActions: Array<TransportAction>?
        val deviceCapabilities: DeviceCapabilities?
        val mediaInfo: MediaInfo?
        val positionInfo: PositionInfo?
        val transportInfo: TransportInfo?
        val transportSettings: TransportSettings?
    }

    // -------------------------------------------------------------------------------------------
    // - Audio
    // -------------------------------------------------------------------------------------------
    interface IAudioControl : IControl {
        fun setMute(channelName: String, desiredMute: Boolean)
        fun getMute(channelName: String): Boolean
        fun setVolume(channelName: String, desiredVolume: UnsignedIntegerTwoBytes)
        fun getVolume(channelName: String): UnsignedIntegerTwoBytes
    }
}